### DISCUSSION TITLE
How to hide tooltip in icon-only `Button`?

### DISCUSSION THREAD

**Discussion by testforvln99** [carbon-design-system/carbon-components-svelte#1483] [Q&A] [answered]
When I use Button components like <Button icon="{PlayFilled}"  iconDescription="start"/>, I found that the iconDescription always pop up. In issue https://github.com/carbon-design-system/carbon/issues/11170#issue-1196077413, @jnm2377 told us that "you can also choose not to include the icon description so the title tooltip doesn't pop up", but the icon-only button must have icon description. 
So how can I forbid  iconDescription popping up from icon-only button?

**Comment by metonym** [comment, marked as answer]
In that issue, the scenario refers to the native `title` attribute, which is controlled by browsers.

For the icon-only Button, you can use global CSS that hides the tooltip text. Note that it's important to specify an icon description for accessibility reasons.

[Svelte REPL](https://svelte.dev/repl/55ac2bd209b64c05b89db27fee23bfc6?version=3.50.0)

```html
<Button iconDescription="Tooltip text" icon={Add} />

<style>
  :global(.bx--btn--icon-only:hover .bx--assistive-text) {
    display: none !important;
  }
	
  :global(.bx--btn.bx--btn--icon-only.bx--tooltip__trigger.bx--tooltip--a11y::before) {
    display: none !important;
  }
</style>
```

**Comment by testforvln99** [comment]
You really save my time! Thanks for you answer.

**Reply by mbgower** [reply]
@testforvln99 note that if the tooltip doesn't appear, no one can discover what the name/purpose of an icon button is. It is unusual _not_ to have a tooltip appear (both on hover and on focus). You may have a good use case, but I wanted to flag.

**Comment by metonym** [comment]
Hiding the tooltip for an icon-only button is now supported: https://github.com/carbon-design-system/carbon-components-svelte/pull/2265

Docs: https://svelte.carbondesignsystem.com/components/Button#icon-only-button-hidden-tooltip
